R/quosure-eval_code.R

```diff
@@ -8,37 +8,35 @@
 #'
 #' @param object (`Quosure`)
 #' @param code (`character` or `language`) code to evaluate. Also accepts and stores comments
-#' @param name (`character(1)`) name of the code block.
 #'
 #' @examples
 #' q1 <- new_quosure(env = list2env(list(a = 1)), code = quote(a <- 1))
-#' q2 <- eval_code(q1, "library(checkmate)")
-#' q3 <- eval_code(q2, "assert_number(a)")
+#' q2 <- eval_code(q1, quote(library(checkmate)))
+#' q3 <- eval_code(q2, quote(assert_number(a)))
 #'
 #' @export
-setGeneric("eval_code", function(object, code, name = "code") {
+setGeneric("eval_code", function(object, code) {
   standardGeneric("eval_code")
 })
 
 #' @rdname eval_code
 #' @export
-setMethod("eval_code", signature = c("Quosure", "character"), function(object, code, name) {
-  checkmate::assert_string(name)
-  if (is.null(names(code))) {
-    code <- paste(code, collapse = "\n")
-    names(code) <- name
-  }
+setMethod("eval_code", signature = c("Quosure", "character"), function(object, code) {
+  eval_code(object, code = parse(text = code, keep.source = FALSE))
+})
+
+#' @rdname eval_code
+#' @export
+setMethod("eval_code", signature = c("Quosure", "expression"), function(object, code) {
   id <- sample.int(.Machine$integer.max, size = length(code))
 
   object@id <- c(object@id, id)
-  object@code <- .keep_code_name_unique(object@code, code)
+  object@env <- rlang::env_clone(object@env, parent = parent.env(.GlobalEnv))
+  object@code <- c(object@code, code)
```

Comment:
so here I would love to have the following:
```
new_object <- clone(object) # new method in which we are doing rlang::env_clone etc.
new_object@id <- c(new_object@id, id)
new_object@code <- c(...)
eval(...)
```

This new (private) method is then used here in eval and also in join. It makes it more explicit that new object is being created and then further modified

What's your view on this?

Reply:
I assume that `clone` wouldn't clone the whole `object` but only the environment. `object` is cloned automatically by R internals when modified. Sounds too much for me

Reply:
ok let's leave it then